core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -67,23 +73,94 @@ public WrapperQueryBuilder(byte[] source, int offset, int length) {
     }
 
     /**
+     * Creates a query builder given a query provided as a bytes array
+     */
+    public WrapperQueryBuilder(byte[] source) {
+        this(source, 0, source.length);
+    }
+
+    /**
      * Creates a query builder given a query provided as a {@link BytesReference}
      */
     public WrapperQueryBuilder(BytesReference source) {
-        this.source = source.array();
+        this.source = source.toBytes();
         this.offset = source.arrayOffset();
         this.length = source.length();
     }
 
+    public byte[] source() {
+        return this.source;
+    }
+
+    public int offset() {
+        return this.offset;
+    }
+
+    public int length() {
+        return this.length;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("query", source, offset, length);
+        printBoostAndQueryName(builder);
```

Comment:
this was not there on purpose, it doesn't really make sense for this wrapper to have its own boost and query _name I think. I was considering adding no-op setters for those like I did in `QueryFilterBuilder`